internal/repository/targets.go

```diff
@@ -73,7 +73,7 @@ func (r *Repository) InitializeTargets(ctx context.Context, signer sslibdsse.Sig
 
 // AddDelegation is the interface for the user to add a new rule to gittuf
 // policy.
-func (r *Repository) AddDelegation(ctx context.Context, signer sslibdsse.SignerVerifier, targetsRoleName string, ruleName string, authorizedKeys []*tuf.Key, rulePatterns []string, signCommit bool) error {
+func (r *Repository) AddDelegation(ctx context.Context, signer sslibdsse.SignerVerifier, targetsRoleName string, ruleName string, authorizedKeys []*tuf.Key, rulePatterns []string, threshold int, signCommit bool) error {
```

Comment:
This list is starting to get long 😵 

We should start thinking about when variadic options with sensible defaults may make more sense ([example](https://pkg.go.dev/github.com/sigstore/sigstore/pkg/signature#SignOption)). 🤔

Reply:
This makes sense but I think we should tackle this separately.